instrumentation/pekko/pekko-http-1.0/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/pekkohttp/v1_0/server/route/RouteConcatenationInstrumentation.java

```diff
@@ -39,8 +58,10 @@ public static void onEnter() {
     }
 
     @Advice.OnMethodExit(onThrowable = Throwable.class, suppress = Throwable.class)
-    public static void onExit() {
-      PekkoRouteHolder.restore();
+    public static void onExit(
+        @Advice.Argument(value = 2) RequestContext ctx,
+        @Advice.Return(readOnly = false) Future<RouteResult> fut) {
+      fut = fut.andThen(new OnExitFinalizer(), ctx.executionContext());
```

Comment:
I need to have a deeper think about the choice of `andThen` here--`andThen` will swallow any exceptions that occur within the PartialFunction which I don't think is what we want.

Reply:
Actually, exceptions raised within the future will still be logged by the Exception Logger. Resolving this.